src/ringmesh/geomodel/geomodel_builder_file.cpp

```diff
@@ -46,26 +47,69 @@
 namespace RINGMesh
 {
     template < index_t DIMENSION >
-    GeoModelBuilderFile< DIMENSION >::GeoModelBuilderFile(
+    GeoModelBuilderFileBase< DIMENSION >::GeoModelBuilderFileBase(
         GeoModel< DIMENSION >& geomodel, std::string filename )
         : GeoModelBuilder< DIMENSION >( geomodel ),
           filename_( std::move( filename ) )
     {
     }
 
     template < index_t DIMENSION >
-    void GeoModelBuilderFile< DIMENSION >::build_geomodel()
+    GeoModelBuilderFileBase< DIMENSION >::~GeoModelBuilderFileBase()
+    {
+    }
+
+    template < index_t DIMENSION >
+    void GeoModelBuilderFileBase< DIMENSION >::build_geomodel()
     {
         if( find_geomodel_dimension( filename_ ) != DIMENSION )
         {
             throw RINGMeshException(
                 "I/O", "Dimension of the GeoModel does not match the file" );
         }
         load_file();
+    }
+
+    GeoModelBuilderFile< 2 >::GeoModelBuilderFile(
+        GeoModel2D& geomodel, std::string filename )
+        : GeoModelBuilderFileBase< 2 >( geomodel, filename )
+    {
+    }
+
+    GeoModelBuilderFile< 2 >::~GeoModelBuilderFile()
+    {
+    }
+
+    void GeoModelBuilderFile< 2 >::build_geomodel()
+    {
+        GeoModelBuilderFileBase< 2 >::build_geomodel();
+        if( this->geomodel_.nb_surfaces() == 0 )
+        {
+            this->build_surfaces_from_corners_and_lines();
+        }
```

Comment:
I am not convinced by this design. I do not see the purpose of a GMBuilderFileBase. Moreover, I would prefer explicitly call functions like the one you just added then automatically call it if needed. I would not like magic builder.